common/cards/single-use/lantern.ts

```diff
@@ -40,9 +41,17 @@ class LanternSingleUseCard extends SingleUseCard {
 				},
 				onResult(modalResult) {
 					if (!modalResult) return 'FAILURE_INVALID_DATA'
+					if (!modalResult.cards) return 'SUCCESS'
+					if (modalResult.cards.length !== 2) return 'FAILURE_INVALID_DATA'
 
-					player.pile.filter((c) => !modalResult.cards.includes(c))
-					modalResult.cards.reverse().map((c: CardT) => player.hand.push(c))
+					const cards: Array<CardT> = modalResult.cards
+
+					player.pile = player.pile.filter((c) => {
+						if (cards.some((d) => c.cardInstance === d.cardInstance)) return false
+						return true
+					})
+
+					cards.map((c) => player.hand.push(c))
```

Comment:
This should be forEach
```suggestion
					cards.forEach((c) => player.hand.push(c))
```